src/graphapi/find_engine.cpp

```diff
@@ -36,12 +36,148 @@
 #include <miopen/graphapi/rng.hpp>
 #include <miopen/graphapi/util.hpp>
 #include <miopen/graphapi/variant_pack.hpp>
+#include <miopen/graphapi/convolution.hpp>
+#include <miopen/graphapi/conv_bias_res_add_activ_forward_executor.hpp>
 
 namespace miopen {
 namespace graphapi {
 
 GraphPatternMatcher::~GraphPatternMatcher() = default;
 
+class ConvBiasResAddActive_Fwd_Pattern : public GraphPatternMatcher
+{
+    static const OpGraph& getPatternGraph()
+    {
+        static auto graph_gen =
+            PatternGraphGenerator::Make({{"OP_CONVOLUTION_FORWARD", {"X", "W"}, {"T_C_0"}},
+                                         {"OP_POINTWISE:ADD", {"T_C_0", "Z"}, {"T_A_0"}},
+                                         {"OP_POINTWISE:ADD", {"T_A_0", "BIAS"}, {"T_A_1"}},
+                                         {"OP_POINTWISE:RELU_FWD", {"T_A_1"}, {"Y"}}});
+
+        return graph_gen->graph();
+    }
+
+    static bool IsBiasNode(OperationPointwise* addNode)
+    {
+        auto& lengthsToCheck  = addNode->getX()->isVirtual() ? addNode->getB()->GetLengths()
+                                                             : addNode->getX()->GetLengths();
+        int notOneOrZeroCount = 0;
+        for(auto& length : lengthsToCheck)
+        {
+            if(length > size_t{1})
+            {
+                notOneOrZeroCount++;
+            }
+        }
+
+        // Assume bias node when at most one dimension is larger than 1.
+        return notOneOrZeroCount <= 1;
+    }
+
+public:
+    static std::unique_ptr<GraphPatternMatcher> Make()
+    {
+        return std::make_unique<ConvBiasResAddActive_Fwd_Pattern>();
+    }
+
+    std::string_view name() const final
+    {
+        static const char* n = "convbiasresaddactivation_fwd";
+        return n;
+    }
+
+    bool matches(const OpGraph* graph_ptr) const final
+    {
+        assert(graph_ptr);
+        return isIsomorphic(*graph_ptr, getPatternGraph());
+    }
+
+    std::vector<Engine> getEngines(OpGraph* graph_ptr) const override
+    {
+        assert(graph_ptr);
+        assert(matches(graph_ptr));
+        auto& graph = *graph_ptr;
+
+        auto* conv = dynamic_cast<OperationConvolutionForward*>(
+            graph.findOutNeighByName(graph.getSourceNode(), "OP_CONVOLUTION_FORWARD"));
+        assert(conv);
+
+        std::size_t in_c  = conv->getX()->GetLengths()[1];
+        std::size_t wei_c = conv->getW()->GetLengths()[1];
+
+        if(wei_c == std::size_t{0})
+        {
+            MIOPEN_THROW(
+                miopenStatusBadParm,
+                "invalid weight tensor provided for graph matching ConvBiasResAddActive pattern");
+        }
+        else if(in_c % wei_c != std::size_t{0})
+        {
+            MIOPEN_THROW(miopenStatusBadParm,
+                         "invalid group count from input and weight tensor for graph matching "
+                         "ConvBiasResAddActive pattern");
+        }
+
+        int groupCount = in_c / wei_c;
```

Comment:
Should these checks be part of the matches check?
Not sure if matches implies that it should run without issue, or just if the pattern matches. 

In this case, I think it makes sense not to be part of the matches check since this implies invalid tensors were passed in for the ConvolutionOp. 

(Looking for feedback from reviewers on this)

Reply:
some of these checks are in the high-level convolution API but then fused convolutions don't seem to go through that API and hence bypass those checks. I don't think we need to check extensively here and the solver should do more checks of the nature of valid params. Here we primarily check the graph (nodes and tensors). This is my initial opinion subject to change.

Reply:
In this specific case, since I need to calculate groupCount for the convolution descriptor , I think I need to check and throw here, but I like the general opinion of how we should handle verification. That will minimize duplicating existing validity checks, and seems like a good approach to me.

Reply:
I guess one you've found a list of applicable engines, you have to go through that list and check `isApplicable` method.
I'm a bit out of the context and probably it's not possible at this step, but we can't return non-applicable solver\engine in any case.